### PR TITLE
refactor: change worker context initializer to bean post processor

### PR DESCRIPTION
**Pull request by rami-nk**
**Description**
This shows how #109 can be refactored.
The BeanPostProcessor interface is implemented to hook into the spring bean lifecycle. 
This interface provides functions that get called for every bean. So instead of iterating over all beans, here we just check in the method that gets called for every bean if it has a specific annotation. This bean than gets mapped and added to a Registry. After this has happened, the registry should contain all mapped annotated bean data. The registry is now queried and we call the same functions on all data points as before. 
For now it does this only for the worker tho. 

**Checklist**

* Code is easy to understand
* No obvious errors or bugs
* CI/CD Pipelines did run successfully

### CODE REVIEW

**Comment by rami-nk** [comment]
@dominikhorn93 we talked about this last time. I created this branch to test it out. Not sure if it is exactly what you were thinking of. Feedback is appreciated

**Comment by dominikhorn93** [comment]
@rami-nk I guess this one has a lot of merge conflicts with https://github.com/FlowSquad/miranum-connect/pull/82 

@lmoesle when do you plan to complete the PR?

File: connect/connect-core/connect-worker/worker-api/src/main/java/io/miragon/miranum/connect/worker/impl/WorkerInfoRegistry.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package io.miragon.miranum.connect.worker.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class WorkerInfoRegistry {
```
Comment (dominikhorn93) [inline connect/connect-core/connect-worker/worker-api/src/main/java/io/miragon/miranum/connect/worker/impl/WorkerInfoRegistry.java:6]: Would name it WokerRegistry, even if it contains a WorkerInfo object

File: connect/connect-core/connect-worker/worker-api/src/main/java/io/miragon/miranum/connect/worker/impl/WorkerInfoRegistry.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    private final List<WorkerInfo> workerInfos = new ArrayList<>();
+
+    public void addWorkerInfo(WorkerInfo workerInfo) {
+        workerInfos.add(workerInfo);
```
Comment (dominikhorn93) [inline connect/connect-core/connect-worker/worker-api/src/main/java/io/miragon/miranum/connect/worker/impl/WorkerInfoRegistry.java:11]: I think the WorkerRegistry is something the WorkerExecuterApi in the PR of @lmoesle should use.

In my point of view we should first merge the PR of @lmoesle.

**Review by dominikhorn93 — COMMENTED** [review]
Great Work 👍

**Pull request closed without merging**